torch/csrc/jit/tensorexpr/codegen.cpp

```diff
@@ -30,11 +30,7 @@ RegisterCodeGenList::StmtFactoryMethod RegisterCodeGenList::
 void RegisterCodeGenList::AddStmtFactoryMethod(
     const std::string& name,
     const StmtFactoryMethod& stmt_factory_method) {
-  auto insert_ret =
-      stmt_factory_methods_.insert(std::make_pair(name, stmt_factory_method));
-  if (!insert_ret.second) {
-    throw std::runtime_error("Duplicated CodeGen names: " + name);
-  }
+  stmt_factory_methods_[name] = stmt_factory_method;
```

Comment:
Interesting. So the global Register are reconstructed, but the register through Meyer's singleton are shared. It is worth looking into why that happens. Nevertheless, this diff seems good.

Reply:
BTW, will it be safer if we keep the first one, and ignore the subsequent ones. I am thinking whether race condition could happen here....

Reply:
I think it's probably OK either to keep the first or keep the new.  Maybe a slight preference this way for simplicity?  (I don't know whether this can race and we'd probably be screwed either way :-) ).